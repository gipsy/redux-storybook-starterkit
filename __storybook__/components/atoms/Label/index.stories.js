import React from 'react'
import { storiesOf } from '@storybook/react'
import Label from '~/src/components/atoms/Label'

storiesOf('Label', module)
  .add('default', () => (
    <Label>Hello</Label>
  ))
  .add('reverse', () => (
    <Label reverse>Hello</Label>
  ))
  .disable(enableFn => {false})
