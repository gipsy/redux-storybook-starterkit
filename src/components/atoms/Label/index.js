import PropTypes from 'prop-types'
import styled from 'styled-components'
import { font, palette } from 'styled-theme'

const Label = styled.label`
  font-family: ${font('primary')};
  color: ${palette('grayscale', 1)};
  margin-bottom: 25px;
  display: inline-block;
  font-size: 1.25rem;
  line-height: 1em;
`

Label.propTypes = {
  reverse: PropTypes.bool,
}

export default Label
