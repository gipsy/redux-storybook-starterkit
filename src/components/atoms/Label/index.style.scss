.Label {
  font-family: ${font('primary')};
  color: ${palette('grayscale', 1)};
  font-size: 1.25rem;
  line-height: 2em;
}
