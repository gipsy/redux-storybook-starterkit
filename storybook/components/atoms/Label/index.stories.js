import React from 'react'
import { storiesOf } from '@storybook/react'
import Label from '~/src/components/atoms/Label'

storiesOf('Components.Atoms', module)
  .add('Label', () => (
    <Label>Hello</Label>
  ))
  .add('reverse', () => (
    <Label reverse>Hello</Label>
  ))
  .disable(enableFn => {false})
