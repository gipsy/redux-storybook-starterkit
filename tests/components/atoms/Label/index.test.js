import React from 'react'
import { shallow } from 'enzyme'
import Label from '~/src/components/atoms/Label'

const wrap = (props = {}) => shallow(<Label {...props} />)

it('renders children when passed in', () => {
  const wrapper = wrap({ children: 'test' })
  expect(wrapper.contains('test')).toBe(true)
})

it('renders props when passed in', () => {
  const wrapper = wrap({ htmlFor: 'foo' })
  expect(wrapper.find({ htmlFor: 'foo' })).toHaveLength(1)
})

it('should render a label', () => {
  const wrapper = wrap({ children: 'Hello Jest!' })
  expect(wrapper).toMatchSnapshot()
})
