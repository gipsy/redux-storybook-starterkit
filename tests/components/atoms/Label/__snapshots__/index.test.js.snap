// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a label 1`] = `
<label
  className="s1cupmrt-0 eccKvY"
>
  Hello Jest!
</label>
`;
